Moved HtmlAttributeIRNode and children from renderer to writer

ajaybhargavb opened the pull request:
@NTaylorMullen @rynowak 

Added an internal abstract `HtmlAttributeWriter`. Will add tests once design is approved.

NTaylorMullen commented on src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/HtmlAttributeWriter.cs:
```diff
@@ -0,0 +1,14 @@
+
+namespace Microsoft.AspNetCore.Razor.Language.CodeGeneration
+{
+    internal abstract class HtmlAttributeWriter
```
Any reason the `WriteHtmlAttribute` shouldn't also be on here?

ajaybhargavb replied:
This is an internal class. Don't want to expose a way for people to override the HtmlAttribute code generation?

ajaybhargavb replied:
Actually I think we could rename this to `HtmlAttributeValueWriter`.

rynowak replied:
All of this should be on BasicWriter

ajaybhargavb commented:
:up: 📅 
- Removed RenderingConventions
- Added Redirect writers
- Updated an integration test to verify redirect scenario for Html attributes.
- Cleanup

ajaybhargavb commented:
🆙 📅 No longer in design.

- Addressed feedback
- Added tests

NTaylorMullen commented on test/Microsoft.AspNetCore.Razor.Language.Test/CodeGeneration/TagHelperHtmlAttributeRuntimeBasicWriterTest.cs:
```diff
@@ -0,0 +1,135 @@
+                ignoreLineEndingDifferences: true);
+        }
+
+        private static CSharpRenderingContext GetCSharpRenderingContext(BasicWriter writer)
```
Any possibility to make a test base class to hold all these infrastructure bits? It looks like they're duplicated quite a bit

ajaybhargavb replied:
I don't think it's going to add a lot of value. `GetCSharpRenderingContext` does different things for BasicWriter and TagHelperWriter. Only `Lower` can be shared. Is it worth it? I don't know........

NTaylorMullen replied:
Ok fair enough.
